### ISSUE TITLE
[Proposal] Nodejs build targets

### ISSUE THREAD

**Issue by Toxicable** [angular/devkit#595]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] feature request
```
### Area
```
- [x] devkit
```
## TLDR
Build targets for Nodejs apps - including auto process restarting, bundling and interacting with Universal apps

## Summary
We should have build targets that can handle building up Nodejs apps.
While this is slightly out of Angulars domain there are some cross overs.
Universal has to be served from a Nodejs app, therefor to provide the best tooling this is a prerequisite.
Since we provide a Monorepo setup already this would be an excellent addition to be able to use one toolchain to handle an entire suite of apps rather than just FE ones

## Binary
A simple target which can compile TS sources and start a nodejs process with args at the entrypoint.
This will also include a watch mode.
Environments would also be a good feature.

## Build
Bundles TS sources into a single bundle.
Optionally include as many external dependancies and produce a package.json that will install the ones that cannot be packaged

## Jasmine
Compiles TS sources and runs jasmine over the output.

## Universal Dev Server
Brings `Binary` and `dev-server` together to make a target which can compile the frontend and backend in watch mode to make it easier for developing against Universal

Inspiration taken from https://github.com/bazelbuild/rules_nodejs

**Comment by alexeagle** [comment]
If it's inspired by Bazel, what if we just suggest that users with this use case use Bazel instead?
